lib/improver/ensemble_copula_coupling/ensemble_copula_coupling_utilities.py

```diff
@@ -0,0 +1,180 @@
+        percentiles = sorted(percentiles)
+    else:
+        msg = "The {} sampling option is not yet implemented.".format(
+            sampling)
+        raise ValueError(msg)
+    return percentiles
+
+
+def create_cube_with_percentiles(percentiles, template_cube, cube_data):
+    """
+    Create a cube with a percentile coordinate based on a template cube.
+
+    Parameters
+    ----------
+    percentiles : List
+        Ensemble percentiles.
+    template_cube : Iris cube
+        Cube to copy majority of coordinate definitions from.
+    cube_data : Numpy array
+        Data to insert into the template cube.
+        The data is expected to have the shape of
+        percentiles (0th dimension), time (1st dimension),
+        y_coord (2nd dimension), x_coord (3rd dimension).
+
+    Returns
+    -------
+    String
+        Coordinate name of the matched coordinate.
+
+    """
+    def _append_to_aux_coords_and_dims(coord_name, aux_coords_and_dims):
+        """
+        Try to append a tuple containing a desired auxiliary coordinate,
+        if the auxiliary coordinate is present on the template cube.
+
+        Parameters
+        ----------
+        coord_name : String
+            The name of the desired auxiliary coordinate.
+        aux_coords_and_dims : List of tuples
+            List of format: [(aux_coord1, dim_coord_to_be_associated_with),
+                             (aux_coord2, dim_coord_to_be_associated_with)]
+            For example: [(forecast_period, 1), (forecast_reference_time, 1)]
+
+        """
+        try:
+            coord = template_cube.coord(coord_name)
+            for coord_tuple in dim_coords_and_dims:
+                if coord_tuple[0].name() in ["time"]:
+                    time_dim = coord_tuple[1]
+                    break
+            aux_coords_and_dims.append((coord, time_dim))
+        except CoordinateNotFoundError:
+            pass
+
+    percentile_coord = iris.coords.DimCoord(
+        np.float32(percentiles), long_name="percentile",
+        units=unit.Unit("1"), var_name="percentile")
+
+    # Aim to create a list of tuples for setting the dim_coords_and_dims
+    # required for a cube. The "realization" or "probability_above_threshold"
+    # coordinates on the cube are ignored, with all other coordinates being
+    # added to the dim_coords_and_dims list. The percentile coordinate tuple
+    # is prepended to this list.
+    dim_coords = []
+    dims = []
+    index = 1
+    for coord in template_cube.dim_coords:
+        if coord.name() in ["realization", "probability_above_threshold"]:
+            continue
+        dim_coords.append(coord)
+        dims.append(index)
+        index += 1
+
+    dim_coords_and_dims = []
+    for coord, dim in zip(dim_coords, dims):
+        dim_coords_and_dims.append((coord, dim))
+    dim_coords_and_dims = [(percentile_coord, 0)] + dim_coords_and_dims
+
+    aux_coords_and_dims = []
+    _append_to_aux_coords_and_dims(
```

Comment:
This is quite specific to the auxillary coordinates we normally have associated with time, maybe if you do the approach below this won't be needed.